src/OpenTelemetry.AutoInstrumentation.StartupHook/StartupHook.cs

```diff
@@ -164,4 +165,45 @@ private static string GetEnvironmentVariable(string variableName)
             return null;
         }
     }
+
+    private static bool EvaluateOpenTelemetryReference(string loaderAssemblyLocation)
+    {
+        var appReferenceAssemblies = Assembly.GetEntryAssembly().GetReferencedAssemblies();
+        var index = Array.FindIndex(appReferenceAssemblies, assembly => assembly.Name == "OpenTelemetry");
```

Comment:
What would happen in the case of references to old versions of `OpenTelemetry.Api`? It seems a possible case: some library is instrumented with OTel but it doesn't reference the SDK.

Reply:
OpenTelemetry.dll will have an indirect reference to `OpenTelemetry.Api`. If we check for OpenTelemetry it should take care of `OpenTelemetry.Api`.

As called out PR description, later we might need to extend and have a generic solution for other OpenTelemetry Instrumentation package and System.Diagnostics.DiagnosticSource for the same issue.